src/tests/system/tests/test_memcache.py

```diff
@@ -1,1660 +1,541 @@
+    assert (
+        result.memberof(["group1", "group2"]) is not None
+    ), "Group membership should still be resolved after SSSD is stopped."
 
-    check_initgroups()
+    assert client.tools.getent.group("group1") is not None, "`group1` should still be found after SSSD is stopped"
 
-    assert client.tools.getent.group(12345) is None, "Group with gid 12345 was found which is not expected"
-    assert client.tools.getent.group(123450) is None, "Group with gid 123450 was found which is not expected"
-    assert client.tools.getent.group("group1") is None, "Group group1 was found which is not expected"
-    assert client.tools.getent.group("group1_") is None, "Group group1_ was found which is not expected"
+    assert client.tools.getent.group("group2") is not None, "`group2` should still be found after SSSD is stopped"
 
 
 @pytest.mark.importance("high")
 @pytest.mark.cache
-@pytest.mark.topology(KnownTopologyGroup.AnyProvider)
-def test_memcache__lookup_initgroups_without_change_in_membership(client: Client, provider: GenericProvider):
+@pytest.mark.topology(KnownTopology.LDAP)
+@pytest.mark.topology(KnownTopology.IPA)
+@pytest.mark.parametrize("case_sensitive", [True, False], ids=["case_sensitive", "case_insensitive"])
+def test_memcache__case_sensitivity_affects_cache_lookup(client: Client, provider: IPA | LDAP, case_sensitive: bool):
     """
-    :title: Invalidated cache, after refresh and stopped SSSD, has everything loaded in memory
+    :title: Cache lookup respects or ignores case based on `case_sensitive`
     :setup:
-        1. Add 'user1' to SSSD
-        2. Set user gid and uid
-        3. Add 'group1' to SSSD
-        4. Set group gid
-        5. Add members to the group
-        6. Start SSSD
+        1. Create user `user1`.
+        2. Set `case_sensitive` in SSSD configuration.
+        3. Start SSSD.
     :steps:
-        1. Find user with id(name) and id(uid)
-        2. Check that user is member of correct groups
-        3. Find group with getent.group(name) and getent.group(gid)
-        4. Check that the group have correct name and gid
-        5. Invalidate whole cache
-        6. Find user with id(name) and id(uid)
-        7. Check that user is member of correct groups
-        8. Find group with getent.group(name) and getent.group(gid)
-        9. Check that the group have correct name and gid
-        10. Stop SSSD
-        11. Find user with id(name) and id(uid)
-        12. Check that user is member of correct groups
-        13. Find group with getent.group(name) and getent.group(gid)
-        14. Check that the group have correct name and gid
+        1. Lookup user1 with mixed casing to populate memory cache.
+        2. Stop SSSD and lookup again with the same and other casings.
+            `case_sensitive = true`: Lookup works only for matching case
+            `case_sensitive = false`: Lookup works only for last case
```

Comment:
What does 'last' mean here?

Reply:
In general, I do no understand this test.

At least I don't find a lookup with "mixed casing" (or I don't understand it properly).